rotkehlchen/chain/zksync_lite/manager.py

```diff
@@ -494,34 +494,37 @@ def get_db_transactions(
         """Gets any zksynclite transactions from the DB depending on the given filter"""
         transactions = []
         with self.database.conn.read_ctx() as cursor:
-            cursor.execute(
+            zksync_lite_data = cursor.execute(
                 f'SELECT identifier, tx_hash, type, timestamp, block_number, from_address, '
                 f'to_address, asset, amount, fee FROM zksynclite_transactions{queryfilter}',
                 bindings,
+            ).fetchall()
+
+            cursor.execute(
+                f'SELECT tx_id, from_asset, from_amount, to_asset, to_amount '
+                f'FROM zksynclite_swaps WHERE tx_id IN '
+                f'(SELECT identifier FROM zksynclite_transactions{queryfilter})',
+                bindings,
             )
-            for entry in cursor:
-                swap_result = None
-                try:
-                    tx = ZKSyncLiteTransaction.deserialize_from_db(entry[1:])
-                    if tx.tx_type == ZKSyncLiteTXType.SWAP:
-                        cursor.execute(
-                            'SELECT from_asset, from_amount, to_asset, to_amount '
-                            'FROM zksynclite_swaps WHERE tx_id=?', (entry[0],),
-                        )
-                        if (swap_result := cursor.fetchone()) is None:
-                            log.error(
-                                f'Could not deserialize zksync lite transaction from the DB due to not finding swap data for tx_id: {entry[0]}',  # noqa: E501
-                            )
-                            continue
+            swap_data = {row[0]: row[1:] for row in cursor}
 
-                        tx.swap_data = ZKSyncLiteSwapData.deserialize_from_db(swap_result)
+        for entry in zksync_lite_data:
+            swap_result = None
+            try:
+                tx = ZKSyncLiteTransaction.deserialize_from_db(entry[1:])
+                if tx.tx_type == ZKSyncLiteTXType.SWAP:
+                    if (tx_id := entry[0]) not in swap_data:
+                        log.error(f'Could not deserialize zksync lite transaction from the DB due to not finding swap data for tx_id: {tx_id}')  # noqa: E501
+                        continue
```

Comment:
It is cleaner if you do

```suggestion
                    if (tx_data := swap_data.get(entry[0]) is None:
                        log.error(f'Could not deserialize zksync lite transaction from the DB due to not finding swap data for tx_id: {entry[0]}')  # noqa: E501
                        continue
```

then later you don't need to do `swap_data[tx_id]`, just `tx_data`